src/workerd/api/global-scope.c++

```diff
@@ -520,6 +552,21 @@ void ServiceWorkerGlobalScope::sendHibernatableWebSocketError(
   KJ_IF_SOME(h, exportedHandler) {
     KJ_IF_SOME(handler, h.webSocketError) {
       auto promise = handler(js, kj::mv(websocket), js.exceptionToJs(kj::mv(e)));
+
+      // If we have a maximum event duration timeout set, we should prevent the actor from running
```

Comment:
I think this code is similar enough to the other call sites and is repeated enough times for us to just make it a private method that takes a `kj::Promise` and returns one too.